core/src/main/java/eu/maveniverse/maven/njord/shared/SessionConfig.java

```diff
@@ -587,10 +623,58 @@ public Optional<String> prefix() {
                 return Optional.ofNullable(prefix);
             }
 
+            @Override
+            public Map<String, Map<String, String>> serverConfigurations() {
+                return serverConfigurations;
+            }
+
+            @Override
+            public Optional<Map<String, String>> serverConfiguration(String id) {
+                return Optional.ofNullable(serverConfigurations.get(id));
+            }
+
             @Override
             public Optional<CurrentProject> currentProject() {
                 return Optional.ofNullable(currentProject);
             }
         }
     }
+
+    static Map<String, Map<String, String>> extractServerConfigurations(Map<String, Object> configProperties) {
+        HashMap<String, Map<String, String>> serverConfigurations = new HashMap<>();
+        configProperties.entrySet().stream()
+                .filter(e -> e.getKey().startsWith("aether.transport.wagon.config.")
```

Comment:
maybe using `Stream.map(<extract server id>, <use value as is>).filter(Objects::nonNull).collect(...)` to populate `serverConfigurations` would be clearer.